.gitlab/workflows/common.yml

```diff
@@ -0,0 +1,26 @@
+#####################
+# Default Variables #
+#####################
+
+stages:
+  - pre_check
+  - build
+  - test
+  - result
+
+variables:
+  ESP_IDF_VERSION: "5.4"
+  ESP_ARDUINO_VERSION: "3.2.1"
```

Comment:
you need to add this file to be updated by `update_version.sh`

Reply:
done